### ISSUE TITLE
Tariffs patch/put handle

### ISSUE THREAD

**Issue by neok** [ocpi/ocpi#172]
Hi guys, as I read from documentation(tariffs) this is an example of some tariff elements:
```
"elements": [{
		"price_components": [{
			"type": "FLAT",
			"price": 2.50,
			"step_size": 1
		}]
	}, {
		"price_components": [{
			"type": "TIME",
			"price": 1.00,
			"step_size": 900
		}],
		"restrictions": {
			"max_power": 32.00
		}
	}, {
		"price_components": [{
			"type": "TIME",
			"price": 2.00,
			"step_size": 600
		}],
		"restrictions": {
			"min_power": 32.00,
			"day_of_week": ["MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY"]
		}
	}
}]
```
How to handle patch request with this data
```{
	"elements": [{
		"price_components": [{
			"type": "TIME",
			"price": 2.50,
			"step_size": 300
		}]
	}]
}
```

So the question is, what should be updated?
If I have two price components with type = `TIME`,  what is the expected result?
Do I need to find price component with type TIME and update it(but we have two)? or do I need to override entire `elements` with this single price component? Or even 3rd case, maybe i should add this new price_component to elements?

**Comment by KevBelisle** [comment]
I believe the "elements" sent in the PATCH request should completely replace the "elements" in the stored tariff.

In other words, you should perform a "shallow merge", not a "deep merge".

**Comment by neok** [comment]
@KevBelisle yep, replaced all elements. thanks

**Comment by RobertdeLeeuw-IHomer** [comment]
PATCH of things in lists is an issue. Especially as things like `price_components` dont have unique IDs that help with a PATCH to replace something.

So for 2.2 we decided to try to explain of limit PATCH. For 3.0 it will probably be removed.
For 2.2 I have removed Tariff PATCH. Tariff is a relative small object and we dont foresee a lot of changes on tariffs. So use PUT instead. For Locations, I have added explanations. Hoping that fixes things. For Session I added an explanation for adding an `charging_period`. For Tokens does not contain a list, so there is no issue, for the moment.
